go/vt/vtgate/engine/simple_projection.go

```diff
@@ -104,8 +108,14 @@ func (sc *SimpleProjection) buildFields(inner *sqltypes.Result) []*querypb.Field
 		return nil
 	}
 	fields := make([]*querypb.Field, 0, len(sc.Cols))
-	for _, col := range sc.Cols {
-		fields = append(fields, inner.Fields[col])
+	for idx, col := range sc.Cols {
+		if sc.ColNames[idx] == "" {
+			fields = append(fields, inner.Fields[col])
+			continue
+		}
+		field := proto.Clone(inner.Fields[col]).(*querypb.Field)
+		field.Name = sc.ColNames[idx]
+		fields = append(fields, field)
```

Comment:
nit: could be simplified
```suggestion
		field := inner.Fields[col]
		if sc.ColNames[idx] != "" {
			field = proto.Clone(field).(*querypb.Field)
			field.Name = sc.ColNames[idx]
		}
		fields = append(fields, field)
```

Reply:
Done